### PR TITLE
Helm! And more goodies.

### PR DESCRIPTION
**Pull request by schottsfired**
*Summary of changes:*
* Helm-based installation \o/
* Custom Helm values.yaml feature with example in `examples/customValues.yaml`
* Updated deployment guide which covers two points above
* K8s 1.14
* Add basic telemetry
* Encrypt EFS OOTB
* Use dedicated `StorageClassName` for EFS (bug fix)
* Move EFS out of workload template, rename `StorageType` to `StorageClassName`
  * Prevents accidental deletion of EFS drive during advanced troubleshooting
* Delegate to CloudBees Core Helm chart for Nginx installation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ci/cloudbees-core-new-vpc.json [hunk 1]
```diff
@@ -5,11 +5,27 @@
...
+    },
+    {
+        "ParameterKey": "CustomValueYaml",
+        "ParameterValue": "override"
     },
     {
-        "ParameterKey": "StorageType",
-        "ParameterValue": "efs"
+        "ParameterKey": "NumberOfMasterNodes",
+        "ParameterValue": "override"
```
Comment (jaymccon) [inline ci/cloudbees-core-new-vpc.json]: We should only be overriding parameter values that are account specific like arn's, and secrets we don't want in git.

File: templates/cloudbees-core-efs.template.yaml [hunk 2]
```diff
@@ -10,6 +10,8 @@ Parameters:
     Type: String
```
Comment (jaymccon) [inline templates/cloudbees-core-efs.template.yaml]: Out of interest, why not use the efs option in the eks qs (largely based on your original template for efs)?

File: templates/cloudbees-core-existing-vpc.template.yaml [hunk 3]
```diff
@@ -769,17 +772,31 @@ Parameters:
...
+    MinValue: 160
+    Default: 160
+    Description:
+      Amount of Amazon EFS provisioned throughput in Mibps. This value is not used 
+      when "gp2" (Amazon EBS) is selected as the storage type. The default value 
+      (160Mibps) is recommended.
+  CustomValueYaml:
+    Type: String
+    Default: ''
+    Description: HTTP(S) or Amazon S3 URL which points to raw yaml containing "custom"
```
Comment (jaymccon) [inline templates/cloudbees-core-existing-vpc.template.yaml]: We should mention that this is optional

File: templates/cloudbees-core-spot-nodes.template.yaml [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# largely based on https://github.com/aws-quickstart/quickstart-amazon-eks/blob/master/templates/amazon-eks-nodegroup.template.yaml
```
Comment (jaymccon) [inline templates/cloudbees-core-spot-nodes.template.yaml]: I would love to get this included in the eks qs for general use,  I'll look at porting it across if you have no objections

File: ci/cloudbees-core-new-vpc.json (hunk 1 above)
Comment (schottsfired) [reply ci/cloudbees-core-new-vpc.json]: Is there another way to pass the `NumberOfMasterNodes` outside of this file? I added it to reduce the size of test deployments, to save $$.

File: templates/cloudbees-core-efs.template.yaml (hunk 2 above)
Comment (schottsfired) [reply templates/cloudbees-core-efs.template.yaml]: I tried, but it doesn't support custom values, so I added them to our version:
```
efsProvisioner.storageClass.name: aws-efs
nodeSelector.partition: masters
```

File: templates/cloudbees-core-existing-vpc.template.yaml (hunk 3 above)
Comment (schottsfired) [reply templates/cloudbees-core-existing-vpc.template.yaml]: Good point! Will do.

File: templates/cloudbees-core-spot-nodes.template.yaml (hunk 4 above)
Comment (schottsfired) [reply templates/cloudbees-core-spot-nodes.template.yaml]: Yay! No objections, as it eliminates maintenance on our side. Thanks!

File: ci/cloudbees-core-new-vpc.json (hunk 1 above)
Comment (jaymccon) [reply ci/cloudbees-core-new-vpc.json]: specifying "override" is just a convention, so if you put the value you'd like for ci in the file, and have an override on your machine the override value will still take precedence

File: templates/cloudbees-core-efs.template.yaml (hunk 2 above)
Comment (jaymccon) [reply templates/cloudbees-core-efs.template.yaml]: I'll get that fixed https://github.com/aws-quickstart/quickstart-amazon-eks/issues/96

File: templates/cloudbees-core-spot-nodes.template.yaml (hunk 4 above)
Comment (jaymccon) [reply templates/cloudbees-core-spot-nodes.template.yaml]: https://github.com/aws-quickstart/quickstart-amazon-eks/issues/97

File: ci/cloudbees-core-new-vpc.json (hunk 1 above)
Comment (schottsfired) [reply ci/cloudbees-core-new-vpc.json]: OK, I see what you mean 👍 

How does https://github.com/aws-quickstart/quickstart-cloudbees-core/pull/21/commits/c870bf5ffdd69195952e4af08daf4c66db217aca look?

**Pull request merged**